File: frontend/types/Task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  deadline: string;
  priority: "low" | "medium" | "high";
  category: string;
  status: "pending" | "in-progress" | "completed";
  created_at: string;
}

export interface TaskStats {
  total: number;
  pending: number;
  inProgress: number;
  completed: number;
  overdue: number;
  dueSoon: number;
  byPriority: {
    low: number;
    medium: number;
    high: number;
  };
  byCategory: Array<{
    category: string;
    count: number;
  }>;
}

File: backend/models/task.py
from datetime import datetime, date
from db import db

class Task(db.Model):
    __tablename__ = 'tasks'
    
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.Text, nullable=False)
    description = db.Column(db.Text)
    deadline = db.Column(db.Date, nullable=False)
    priority = db.Column(db.String(20), nullable=False, default='medium')
    category = db.Column(db.String(100))
    status = db.Column(db.String(20), nullable=False, default='pending')
    created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
    
    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'deadline': self.deadline.isoformat() if self.deadline else None,
            'priority': self.priority,
            'category': self.category,
            'status': self.status,
            'created_at': self.created_at.isoformat() if self.created_at else None
        }
    
    @classmethod
    def from_dict(cls, data):
        return cls(
            title=data.get('title'),
            description=data.get('description'),
            deadline=datetime.strptime(data['deadline'], '%Y-%m-%d').date() if data.get('deadline') else None,
            priority=data.get('priority', 'medium'),
            category=data.get('category'),
            status=data.get('status', 'pending')
        )